### PR TITLE
docs(rfcs): reorder to better fit docs page

### PR DESCRIPTION
**Pull request by gergelyke**

### CODE REVIEW

File: rfcs/button-component.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
...
+  <Button
+    startEnhancer={
+      <StyledStartEnhancer>
+        <SvgIcon />
+      </StyledStartEnhancer>
+    }
+    endEnhancer={
+      <StyledEndEnhancer>
+        <SvgIcon />
+      </StyledEndEnhancer>
```
Comment (schnerd) [inline rfcs/button-component.md]: Unrelated to @gergelyke 's changes but @abmai @kahwee why do we need StyledEnhancer here? Shouldn't it automatically be using these internally already? Seems like you should only need to import `Styled*` when you want to use the `overrides` mechanism.

File: rfcs/button-component.md [hunk 2]
```diff
@@ -1,5 +1,50 @@
 # Button Component
 
+### Usage
+
+#### Basic usage
```
Comment (schnerd) [inline rfcs/button-component.md]: Nice, I like the idea of showing both basic usage and advanced usage.

File: rfcs/button-component.md [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+
+```javascript
+import * as React from 'react';
+import {Button, KIND} from 'baseui/button';
+
+export default () => <Button kind={KIND.primary}>Submit</Button>;
```
Comment (schnerd) [inline rfcs/button-component.md]: Should we still show some sort of basic onClick handler here?

```js
<Button onClick={() => console.log('clicked')} kind={KIND.primary}>
  Submit
</Button>
```

File: rfcs/checkbox-component.md [hunk 4]
```diff
@@ -1,5 +1,75 @@
...
+### Usage
+
+#### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Checkbox} from 'baseui/checkbox';
+
+export default () => <Checkbox />;
+```
```
Comment (schnerd) [inline rfcs/checkbox-component.md]: I think for form controls the basic usage should hopefully still show passing a value in and change handlers

```js
<Checkbox 
  checked={props.isEnabled}
  onChange={evt => props.setEnabled(evt.target.checked)}
>
  Enabled
</Button>
```

File: rfcs/input-component.md [hunk 5]
```diff
@@ -1,133 +1,145 @@
...
+### Usage
+
+#### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Input} from 'baseui/input';
+
+export default () => <Input />;
+```
```
Comment (schnerd) [inline rfcs/input-component.md]: Same here, maybe something like:

```js
<Input 
  value={props.name}
  onChange={evt => props.setName(evt.target.value)}
/>
```

File: rfcs/button-component.md (hunk 1 above)
Comment (abmai) [reply rfcs/button-component.md]: Hm my guess is probably just an oversight in writing these. It should be used internally already yes. cc @kahwee

**Pull request merged**